providers/base/bin/edid_cycle.py

```diff
@@ -17,17 +17,34 @@
 from checkbox_support.scripts.zapper_proxy import (             # noqa: E402
     zapper_run)
 
+PATTERN = "^HDMI-1.*\n.* (\\d+x\\d+).*\\*"
+
 
 def check_resolution():
-    output = subprocess.check_output('xdpyinfo')
-    for line in output.decode(sys.stdout.encoding).splitlines():
-        if 'dimensions' in line:
-            match = re.search(r'(\d+)x(\d+)\ pixels', line)
-            if match and len(match.groups()) == 2:
-                return '{}x{}'.format(*match.groups())
+    """
+    Check output resolution on HDMI using randr.
+    Both gnome-randr and xrandr highlight the currently
+    selected resolution for each monitor with a `*`.
+    """
+    if os.getenv('XDG_SESSION_TYPE') == 'wayland':
+        cmd = "gnome-randr"
+    else:
+        cmd = "xrandr"
+
+    randr_output = subprocess.check_output(
+        [cmd],
+        universal_newlines=True, encoding="utf-8")
+
+    match = re.search(PATTERN, randr_output, re.MULTILINE | re.DOTALL)
+    if match:
+        return match.group(1)
+    return None
 
 
 def change_edid(host, edid_file):
+    """Clear EDID and then 'plug' back a new monitor."""
+    zapper_run(host, "change_edid", None)
+    time.sleep(1)
```

Comment:
I'll be thinking loud here:
If there's a delay that we have to wait in order for EDID to land, it seems like the better place to do that would be on Zapper side. Cause right now it "maybe changes" :)
Second thing is why 1s? IIRC we can check whether they got propagated (`v4l2 (something, something) query timings`), so I guess a busy loop would be better, that after 5seconds returns an error if it failed to do appropriate operation.
What do you think?

Reply:
Good point, those manual delays are not nice :D
It could be done on both sides actually. On Zapper side there's a problem though: the EDID change is immediate but the timings in use are available (and credible) only if there's a monitor connected. The command is `v4l2-ctl --query-dv-timings`.
Since I don't want the Zapper service to hang and timeout only because there's no monitor connected I'd prefer to have the busy loop on the checkbox side. Done in the last commit.